### ISSUE TITLE
Test which was passing with 2.28.0 fails with 2.33.0

### ISSUE THREAD

**Issue by rotty3000** [lukas-krecan/JsonUnit#502]
**Describe the bug**
1. Describe which variant of the API are you using (AssertJ, Hamcrest, ...)
    *Using AssertJ varient*
2. Create a small reproducible example and paste it the the bug report.
    *I not completely sure this is an issue in JSONUnit just yet*
3. Describe expected behavior
    *Test which used to pass should still pass... I think?*

For example:

My test which used to pass fails if I update: See [here](https://github.com/apache/aries-jax-rs-whiteboard/runs/5972926253?check_suite_focus=true#step:4:1000)

test is:
```java
        ...
        
        String result = response.readEntity(String.class);

        JsonAssertions.assertThatJson(
            result
        ).and(
            j -> j.isObject(),
            j -> j.node(
                "bundles"
            ).isArray(
            ).element(
                0
            ).asString(
            ).contains(
                "http://", "/framework/bundle/0"
            )
        );
 ```
Exception is:
```
TEST org.apache.aries.jax.rs.rest.management.test.BundlesTest#getBundlesJSON() <<< ERROR: Can not parse actual value: 'http://localhost:33311/rms/framework/bundle/0'
java.lang.IllegalArgumentException: Can not parse actual value: 'http://localhost:33311/rms/framework/bundle/0'
	at net.javacrumbs.jsonunit.core.internal.AbstractNodeFactory.newParseException(AbstractNodeFactory.java:71)
	at net.javacrumbs.jsonunit.core.internal.JohnzonNodeFactory.readValue(JohnzonNodeFactory.java:76)
	at net.javacrumbs.jsonunit.core.internal.AbstractNodeFactory.readValue(AbstractNodeFactory.java:83)
	at net.javacrumbs.jsonunit.core.internal.AbstractNodeFactory.convertToNode(AbstractNodeFactory.java:38)
	at net.javacrumbs.jsonunit.core.internal.JohnzonNodeFactory.convertToNode(JohnzonNodeFactory.java:40)
	at net.javacrumbs.jsonunit.core.internal.Converter.convertToNode(Converter.java:128)
	at net.javacrumbs.jsonunit.core.internal.JsonUtils.convertToJson(JsonUtils.java:60)
	at net.javacrumbs.jsonunit.core.internal.JsonUtils.convertToJson(JsonUtils.java:45)
	at net.javacrumbs.jsonunit.assertj.JsonAssert.<init>(JsonAssert.java:68)
	at net.javacrumbs.jsonunit.assertj.JsonListAssert.lambda$new$0(JsonListAssert.java:38)
	at org.assertj.core.api.FactoryBasedNavigableListAssert.toAssert(FactoryBasedNavigableListAssert.java:43)
	at org.assertj.core.api.AbstractIterableAssert.internalElement(AbstractIterableAssert.java:3534)
	at org.assertj.core.api.AbstractIterableAssert.element(AbstractIterableAssert.java:3419)
	at org.apache.aries.jax.rs.rest.management.test.BundlesTest.lambda$1(BundlesTest.java:66)
	at net.javacrumbs.jsonunit.assertj.JsonAssert.lambda$and$0(JsonAssert.java:100)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:647)
	at net.javacrumbs.jsonunit.assertj.JsonAssert.and(JsonAssert.java:100)
	at org.apache.aries.jax.rs.rest.management.test.BundlesTest.getBundlesJSON(BundlesTest.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at aQute.tester.bundle.engine.BundleDescriptor.executeChild(BundleDescriptor.java:49)
	at aQute.tester.bundle.engine.BundleEngine.lambda$executeBundle$7(BundleEngine.java:120)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485)
	at aQute.tester.bundle.engine.BundleEngine.executeBundle(BundleEngine.java:120)
	at aQute.tester.bundle.engine.BundleEngine.lambda$execute$5(BundleEngine.java:100)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485)
	at aQute.tester.bundle.engine.BundleEngine.execute(BundleEngine.java:100)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at aQute.tester.junit.platform.Activator.test(Activator.java:439)
	at aQute.tester.junit.platform.Activator.automatic(Activator.java:344)
	at aQute.tester.junit.platform.Activator.run(Activator.java:216)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at aQute.launcher.Launcher.launch(Launcher.java:450)
	at aQute.launcher.Launcher.run(Launcher.java:184)
	at aQute.launcher.Launcher.main(Launcher.java:160)
	at aQute.launcher.pre.EmbeddedLauncher.executeWithRunPath(EmbeddedLauncher.java:170)
	at aQute.launcher.pre.EmbeddedLauncher.findAndExecute(EmbeddedLauncher.java:135)
	at aQute.launcher.pre.EmbeddedLauncher.main(EmbeddedLauncher.java:52)
Caused by: javax.json.stream.JsonParsingException: Unexpected character 'h' (Codepoint: 104) on [lineNumber=1, columnNumber=2, streamOffset=1]. Reason is [[Expected structural character or digit or 't' or 'n' or 'f' or '-']]
	at org.apache.johnzon.core.JsonStreamParserImpl.uexc(JsonStreamParserImpl.java:1049)
	at org.apache.johnzon.core.JsonStreamParserImpl.uexc(JsonStreamParserImpl.java:1055)
	at org.apache.johnzon.core.JsonStreamParserImpl.defaultHandling(JsonStreamParserImpl.java:494)
	at org.apache.johnzon.core.JsonStreamParserImpl.internalNext(JsonStreamParserImpl.java:485)
	at org.apache.johnzon.core.JohnzonJsonParserImpl.next(JohnzonJsonParserImpl.java:56)
	at org.apache.johnzon.core.JsonReaderImpl.readValue(JsonReaderImpl.java:87)
	at net.javacrumbs.jsonunit.core.internal.JohnzonNodeFactory.readValue(JohnzonNodeFactory.java:74)
	... 116 more
```

**Comment by rotty3000** [comment]
The exact value in test was the following:
```json
{"bundles":["http://localhost:33621/rms/framework/bundle/0","http://localhost:33621/rms/framework/bundle/1","http://localhost:33621/rms/framework/bundle/2","http://localhost:33621/rms/framework/bundle/3","http://localhost:33621/rms/framework/bundle/4","http://localhost:33621/rms/framework/bundle/5","http://localhost:33621/rms/framework/bundle/6","http://localhost:33621/rms/framework/bundle/7","http://localhost:33621/rms/framework/bundle/8","http://localhost:33621/rms/framework/bundle/9","http://localhost:33621/rms/framework/bundle/10","http://localhost:33621/rms/framework/bundle/11","http://localhost:33621/rms/framework/bundle/12","http://localhost:33621/rms/framework/bundle/13","http://localhost:33621/rms/framework/bundle/14","http://localhost:33621/rms/framework/bundle/15","http://localhost:33621/rms/framework/bundle/16","http://localhost:33621/rms/framework/bundle/17","http://localhost:33621/rms/framework/bundle/18","http://localhost:33621/rms/framework/bundle/19","http://localhost:33621/rms/framework/bundle/20","http://localhost:33621/rms/framework/bundle/21","http://localhost:33621/rms/framework/bundle/22","http://localhost:33621/rms/framework/bundle/23","http://localhost:33621/rms/framework/bundle/24","http://localhost:33621/rms/framework/bundle/25","http://localhost:33621/rms/framework/bundle/26","http://localhost:33621/rms/framework/bundle/27","http://localhost:33621/rms/framework/bundle/28","http://localhost:33621/rms/framework/bundle/29","http://localhost:33621/rms/framework/bundle/30","http://localhost:33621/rms/framework/bundle/31","http://localhost:33621/rms/framework/bundle/32","http://localhost:33621/rms/framework/bundle/33","http://localhost:33621/rms/framework/bundle/34","http://localhost:33621/rms/framework/bundle/35","http://localhost:33621/rms/framework/bundle/36","http://localhost:33621/rms/framework/bundle/37","http://localhost:33621/rms/framework/bundle/38","http://localhost:33621/rms/framework/bundle/39","http://localhost:33621/rms/framework/bundle/40","http://localhost:33621/rms/framework/bundle/41","http://localhost:33621/rms/framework/bundle/42","http://localhost:33621/rms/framework/bundle/43","http://localhost:33621/rms/framework/bundle/44","http://localhost:33621/rms/framework/bundle/45","http://localhost:33621/rms/framework/bundle/46","http://localhost:33621/rms/framework/bundle/47","http://localhost:33621/rms/framework/bundle/48","http://localhost:33621/rms/framework/bundle/49","http://localhost:33621/rms/framework/bundle/50","http://localhost:33621/rms/framework/bundle/51","http://localhost:33621/rms/framework/bundle/52","http://localhost:33621/rms/framework/bundle/53","http://localhost:33621/rms/framework/bundle/54","http://localhost:33621/rms/framework/bundle/55","http://localhost:33621/rms/framework/bundle/56","http://localhost:33621/rms/framework/bundle/57","http://localhost:33621/rms/framework/bundle/58","http://localhost:33621/rms/framework/bundle/59","http://localhost:33621/rms/framework/bundle/60","http://localhost:33621/rms/framework/bundle/61","http://localhost:33621/rms/framework/bundle/62","http://localhost:33621/rms/framework/bundle/63","http://localhost:33621/rms/framework/bundle/64","http://localhost:33621/rms/framework/bundle/65","http://localhost:33621/rms/framework/bundle/66","http://localhost:33621/rms/framework/bundle/67","http://localhost:33621/rms/framework/bundle/68"]}
```

**Comment by lukas-krecan** [comment]
Hi, thanks for reporting. It's indeed a bug, but it will take me some time to unravel it.

**Comment by lukas-krecan** [comment]
Can you please try 2.34.0?

**Comment by rotty3000** [comment]
I will try it later today. Standby!

**Comment by rotty3000** [comment]
Problem was resolved by 2.34.0! Thanks @lukas-krecan
